### PR TITLE
Group-owned cases should be downloaded when Mobile Workers is selected

### PR DESCRIPTION
**Pull request by kaapstorm**
[FB 166197](http://manage.dimagi.com/default.asp?166197)

Changing this functionality didn't break any tests (at least not hqcase tests). I think a test might be in order. 

buddy @dannyroberts, @amsagoff

### CODE REVIEW

**Comment by czue** [comment]
did you mean to include the submodule update? I think that's what's preventing it from being mergeable

**Comment by kaapstorm** [comment]
Thanks @czue. No I didn't mean that.

File: corehq/apps/hqcase/export.py [hunk 1]
```diff
@@ -57,10 +56,7 @@ def get_matching_owner(case):
             return get_owner_id(case)
 
     def might_be_relevant(case):
-        if filter_group:
-            return get_owner_id(case) in owner_ids
-        else:
-            return not users or get_matching_owner(case)
+        return not filter_group or get_owner_id(case) in owner_ids
```
Comment (dannyroberts) [inline corehq/apps/hqcase/export.py]: This is the same as

```python
         if filter_group:
            return get_owner_id(case) in owner_ids
        else:
            return True
```

right? What's was the logic behind this previous code that's now being changed?

**Comment by dannyroberts** [comment]
@kaapstorm I merged assuming you were sure this was a good change. (If that was wrong, in the future make sure to use the don't merge label!) Still would like to understand the logic behind removing that condition.

File: corehq/apps/hqcase/export.py (hunk 1 above)
Comment (kaapstorm) [inline corehq/apps/hqcase/export.py]: That's right. Previous logic returned True if owners (mobile users) had not been specified (i.e. they were unknown), or, if they had been specified, then only if the case was owned. This contradicted the requirement for [FB 166197](http://manage.dimagi.com/default.asp?166197), where this should also return True for cases not owned by users. 

i.e. `not users or get_matching_owner(case) or not get_matching_owner(case)` == `not users or True` == `True`

If you think I've misinterpreted the bug, please let me know, but I'm pretty certain this is what @amsagoff meant.

**Comment by kaapstorm** [comment]
Thanks @dannyroberts. After the discussion on the previous PR ([8612](https://github.com/dimagi/commcare-hq/pull/6812)) where I chose to keep this logic and add another option instead, it turned out that adding an option was the wrong approach, and that changing this logic would bring it in line with other report filters.

**Comment by dannyroberts** [comment]
Ok, thanks for the explanation. Makes sense!

**Comment by kaapstorm** [comment]
Np. I should have put a longer explanation in the PR comment.

**Pull request merged**